.gitlab-ci.yml

```diff
@@ -146,74 +144,53 @@ build:merge:
     # Runs on tag pipeline where the tag is a prerelease or release version
     - if: $CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
 
-build:linux:
+build:platforms-generate:
   stage: build
   needs: []
   script:
     - >
-      nix-shell --run '
-      npm run build --verbose;
-      '
+        nix-shell --run '
+        ./scripts/build-platforms-generate.sh > ./tmp/build-platforms.yml
+        '
   artifacts:
     when: always
     paths:
-      # Only the build:linux preserves the dist
-      - ./dist
+      - ./tmp/build-platforms.yml
   rules:
     # Runs on staging commits and ignores version commits
     - if: $CI_COMMIT_BRANCH == 'staging' && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
     # Runs on tag pipeline where the tag is a prerelease or release version
     - if: $CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
 
-build:windows:
+build:platforms:
   stage: build
-  needs: []
-  tags:
-    - windows
-  before_script:
-    - mkdir -Force "$CI_PROJECT_DIR/tmp"
-    - choco install nodejs --version=16.14.2 -y
-    - refreshenv
-  script:
-    - npm config set msvs_version 2019
-    - npm install --ignore-scripts
-    - $env:Path = "$(npm bin);" + $env:Path
-    - npm run build --verbose
-    # - npm test -- --ci
-  # artifacts:
-  #   when: always
-  #   reports:
-  #     junit:
-  #       - ./tmp/junit/junit.xml
+  needs:
+    - build:platforms-generate
+  inherit:
+    variables: false
+  trigger:
+    include:
+      - artifact: tmp/build-platforms.yml
+        job: build:platforms-generate
+    strategy: depend
   rules:
     # Runs on staging commits and ignores version commits
     - if: $CI_COMMIT_BRANCH == 'staging' && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
     # Runs on tag pipeline where the tag is a prerelease or release version
     - if: $CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
 
-build:macos:
+build:dist:
```

Comment:
This `build:dist` job which took over the building of the distribution from `build:linux` should be placed ahead of the `build:platforms-generate`.

This allows in the future if we have to pass the `dist` into the child pipelines (or any other build artifacts).

Reply:
To clarify, both check stage and build stage are running the tests.

The difference is that check stage tests only run during the `feature.*` branches`. While the build stage tests run as part of `staging` branch`.

Now why do this? It's mainly because in the check stage, these jobs don't produce artifacts, they only test tings.

But during the `build` stage, these jobs do produce artifacts.

Now because in some cases like in the TS demo lib native, it is necessary to "build" before you run the tests (due to native code), I combined the building and testing together in the `build` stage`.

In the TS demo lib native, the build stage jobs produce both `./prebuilds/` and `./dist` artifacts. Which are then used later for integration and prerelease/release.

So I had previously just kept the `./dist` from the `build:linux` job.

Now that we are changing things around, I believe `build:dist` should occur before `build:platforms-generate`.

However for PK, it's not necessary to connect these jobs together as a dependency. So I'm just saying move the `build:dist` above `build:platforms-generate`. But this doesn't mean to add `build:dist` to the `needs` list of `build:platforms-generate`. They are not just independent jobs that can run in parallel.

Reply:
To clarify, both check stage and build stage are running the tests.

The difference is that check stage tests only run during the `feature.*` branches`. While the build stage tests run as part of `staging` branch`.

Now why do this? It's mainly because in the check stage, these jobs don't produce artifacts, they only test tings.

But during the `build` stage, these jobs do produce artifacts.

Now because in some cases like in the TS demo lib native, it is necessary to "build" before you run the tests (due to native code), I combined the building and testing together in the `build` stage`.

In the TS demo lib native, the build stage jobs produce both `./prebuilds/` and `./dist` artifacts. Which are then used later for integration and prerelease/release.

So I had previously just kept the `./dist` from the `build:linux` job.

Now that we are changing things around, I believe `build:dist` should occur before `build:platforms-generate`.

However for PK, it's not necessary to connect these jobs together as a dependency. So I'm just saying move the `build:dist` above `build:platforms-generate`. But this doesn't mean to add `build:dist` to the `needs` list of `build:platforms-generate`. They are not just independent jobs that can run in parallel.